packages/core/test/financial-templates/common/financial-product-libraries/StructuredNoteFinancialProductLibrary.js

```diff
@@ -0,0 +1,106 @@
+const { didContractThrow, ZERO_ADDRESS } = require("@uma/common");
+const { assert } = require("chai");
+
+// Tested Contract
+const StructuredNoteFinancialProductLibrary = artifacts.require("StructuredNoteFinancialProductLibrary");
+
+// Helper contracts
+const Timer = artifacts.require("Timer");
+const ExpiringMultiPartyMock = artifacts.require("ExpiringMultiPartyMock");
+
+const { toWei, toBN } = web3.utils;
+const strikePrice = toBN(toWei("400"));
+
+contract("StructuredNoteFinancialProductLibrary", function() {
+  let structuredNoteFPL;
+  let expiringMultiParty;
+  let timer;
+  let expirationTime;
+
+  beforeEach(async () => {
+    timer = await Timer.deployed();
+
+    expirationTime = (await timer.getCurrentTime()) + 100; // use 100 seconds in the future as the expiration time.
+    structuredNoteFPL = await StructuredNoteFinancialProductLibrary.new(timer.address);
+    expiringMultiParty = await ExpiringMultiPartyMock.new(structuredNoteFPL.address, expirationTime, timer.address);
+
+    await structuredNoteFPL.setFinancialProductStrike(expiringMultiParty.address, {
+      rawValue: strikePrice.toString()
+    });
+  });
+  it("Strike correctly set", async () => {
+    assert.equal(
+      (await structuredNoteFPL.getStrikeForFinancialProduct(expiringMultiParty.address)).toString(),
+      strikePrice.toString()
+    );
+  });
+  it("Can not re-set the strike for a given financial product", async () => {
+    assert(
+      await didContractThrow(
+        structuredNoteFPL.setFinancialProductStrike(expiringMultiParty.address, {
+          rawValue: strikePrice.toString()
+        })
+      )
+    );
+  });
+  it("Can not set strike price for invalid financial product", async () => {
```

Comment:
This only fails for the ZERO-ADDRESS right? Or does it fail for any contract that doesn't implement `expirationTimestamp`? If the latter, then can you add a test case for a non-zero-address, non-EMP address?

Reply:
it's the latter and I've added a test